### PR TITLE
Add Cloudflare turn servers

### PR DESCRIPTION
**Pull request by Brutus5000**
image

### CODE REVIEW

File: src/main/kotlin/com/faforever/icebreaker/service/cloudflare/ApiKeyAuthenticationRequestFilter.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.faforever.icebreaker.service.cloudflare
+
+import jakarta.annotation.Priority
+import jakarta.ws.rs.Priorities
+import jakarta.ws.rs.client.ClientRequestContext
+import jakarta.ws.rs.client.ClientRequestFilter
+import jakarta.ws.rs.core.HttpHeaders
+
+@Priority(Priorities.AUTHENTICATION)
+class ApiKeyAuthenticationRequestFilter(turnApiKey: String) : ClientRequestFilter {
```
Comment (Sheikah45) [inline src/main/kotlin/com/faforever/icebreaker/service/cloudflare/ApiKeyAuthenticationRequestFilter.kt:10]: I am a little surprised we have to write our own filter for this, but seems like that is the way to do it.

File: src/main/kotlin/com/faforever/icebreaker/service/cloudflare/CloudflareSessionHandler.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.faforever.icebreaker.config.FafProperties
+import com.faforever.icebreaker.service.Server
+import com.faforever.icebreaker.service.Session
+import com.faforever.icebreaker.service.SessionHandler
+import jakarta.inject.Singleton
+
+@Singleton
+class CloudflareSessionHandler(
+    cloudflareProperties: CloudflareProperties,
```
Comment (Sheikah45) [inline src/main/kotlin/com/faforever/icebreaker/service/cloudflare/CloudflareSessionHandler.kt:11]: missing val?

File: src/main/kotlin/com/faforever/icebreaker/service/cloudflare/CloudflareSessionHandler.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        ).let {
+            listOf(
+                Session.Server(
+                    id = SERVER_NAME,
+                    username = it.iceServers.username,
+                    credential = it.iceServers.credential,
+                    urls = it.iceServers.urls.map { url ->
+                        // A sample response looks like "stun:fr-turn1.xirsys.com"
+                        // The java URI class fails to read host and port due to the missing // after the :
+                        // Thus we "normalize" the uri, even though it is technically valid
```
Comment (Sheikah45) [inline src/main/kotlin/com/faforever/icebreaker/service/cloudflare/CloudflareSessionHandler.kt:44]: This is true for cloudflare as well?

**Review by Sheikah45 — APPROVED** [review]

File: src/main/kotlin/com/faforever/icebreaker/service/cloudflare/CloudflareSessionHandler.kt (hunk 3 above)
Comment (Brutus5000) [reply src/main/kotlin/com/faforever/icebreaker/service/cloudflare/CloudflareSessionHandler.kt:44]: Yes. Surprisingly, they also have the exact same response structure for credential and username. Maybe it is part of the RFC

File: src/main/kotlin/com/faforever/icebreaker/service/cloudflare/CloudflareSessionHandler.kt (hunk 2 above)
Comment (Brutus5000) [reply src/main/kotlin/com/faforever/icebreaker/service/cloudflare/CloudflareSessionHandler.kt:11]: val is only required if I access the variable after initialization. This is not the case here, this it is not required

File: src/main/kotlin/com/faforever/icebreaker/service/cloudflare/ApiKeyAuthenticationRequestFilter.kt (hunk 1 above)
Comment (Brutus5000) [reply src/main/kotlin/com/faforever/icebreaker/service/cloudflare/ApiKeyAuthenticationRequestFilter.kt:10]: Not just you!

**Pull request merged**